src/components/Calculator.js

```diff
@@ -1,54 +1,42 @@
-          <button type="button" className="designcolor" onClick={this.updateCal} id="three">3</button>
-          <button type="button" className="designcolor" onClick={this.updateCal} id="two">2</button>
-          <button type="button" className="designcolor" onClick={this.updateCal} id="one">1</button>
-          <button type="button" className="designcolor2" onClick={this.updateCal} id="one">+</button>
-          <button type="button" className="designcolor" onClick={this.updateCal} id="zero">0</button>
-          <button type="button" className="designcolor" onClick={this.updateCal} id="decimalpoint">&sdot;</button>
-          <button type="button" className="designcolor2" onClick={this.updateCal} id="equals">=</button>
-        </div>
+  const updateCal = (e) => {
+    setState((state) => calculate(state, e.target.textContent));
+  };
+  return (
+    <div className="layout">
+      <div className="print">
+        <h3 className="printout">
+          {state.total}
+          {state.operation}
+          {state.next}
+        </h3>
+      </div>
+      <div className="keyboard">
+        <button type="button" className="designcolor" onClick={updateCal} id="Clear">AC</button>
+        <button type="button" className="designcolor" onClick={updateCal} id="plusMinus">+/-</button>
+        <button type="button" className="designcolor" onClick={updateCal} id="percentage">%</button>
+        <button type="button" className="designcolor2" onClick={updateCal} id="divide">&divide;</button>
+        <button type="button" className="designcolor" onClick={updateCal} id="seven">7</button>
+        <button type="button" className="designcolor" onClick={updateCal} id="eight">8</button>
+        <button type="button" className="designcolor" onClick={updateCal} id="nine">9</button>
+        <button type="button" className="designcolor2" onClick={updateCal} id="multiply">&times;</button>
+        <button type="button" className="designcolor" onClick={updateCal} id="six">6</button>
+        <button type="button" className="designcolor" onClick={updateCal} id="five">5</button>
+        <button type="button" className="designcolor" onClick={updateCal} id="four">4</button>
+        <button type="button" className="designcolor2" onClick={updateCal} id="minus">&minus;</button>
+        <button type="button" className="designcolor" onClick={updateCal} id="three">3</button>
+        <button type="button" className="designcolor" onClick={updateCal} id="two">2</button>
+        <button type="button" className="designcolor" onClick={updateCal} id="one">1</button>
+        <button type="button" className="designcolor2" onClick={updateCal} id="one">+</button>
+        <button type="button" className="designcolor" onClick={updateCal} id="zero">0</button>
+        <button type="button" className="designcolor" onClick={updateCal} id="decimalpoint">&sdot;</button>
+        <button type="button" className="designcolor2" onClick={updateCal} id="equals">=</button>
```

Comment:
- [ ] The calculator disappears when I click on the equal sign after multiplication and a rest. Please take a look. 😉